app/views/agents/show.html.erb

```diff
@@ -161,17 +161,31 @@
               <b>Options:</b>
               <pre><%= Utils.pretty_jsonify @agent.options || {} %></pre>
             </p>
-
-            <p id="memory" data-agent-id="<%= @agent.id %>">
-              <b>Memory:</b>
-              <% if @agent.memory.present? %>
-                <i class="fa fa-spinner fa-pulse spinner"></i>
-                <i class="fa fa-trash action-icon clear"></i>
-              <% end %>
-              <pre class="memory"><%= Utils.pretty_jsonify @agent.memory || {} %></pre>
-            </p>
+            <button type="submit" class="btn btn-default" id="target">Reveal Memory</button>
+            <button type="submit" class="btn btn-default" id="target1">Hide Memory</button>
+            <div id="memorytoggle" class="hidden">
+              <p id="memory" data-agent-id="<%= @agent.id %>">
+                <b>Memory:</b>
+                <% if @agent.memory.present? %>
+                  <i class="fa fa-spinner fa-pulse spinner"></i>
+                  <i class="fa fa-trash action-icon clear"></i>
+                <% end %>
+                <pre class="memory"><%= Utils.pretty_jsonify @agent.memory || {} %></pre>
+              </p>
+            </div>
           </div>
         </div>
       </div>
   </div>
 </div>
+
+
+<%= javascript_tag do %>
```

Comment:
New code should go in agent-show-page.js.coffee.  Can bind it at the top.  Also see the jQuery [toggle](http://api.jquery.com/toggle/) method.